e2e/src/Utils.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  BrowserSize,
+  Coordinates,
+  MetaMaskElementLocator,
+  ScreenPercentage,
+} from './types';
+
+class Utils {
+  static getLocatorPerPlatformAndStrategy(
+    locator: MetaMaskElementLocator,
+  ): string {
+    const platformLocator = driver.isIOS
+      ? locator.iosLocator
+      : locator.androidLocator;
+
+    // In case the locator was not provided for the platform it is running on
+    if (platformLocator === undefined) {
+      throw new Error('Platform locator is undefined');
+    }
+
+    if (
+      locator.androidLocator === undefined &&
+      locator.iosLocator === undefined
+    ) {
+      throw new Error('Locator is undefined');
+    }
+
+    if (driver.isIOS && locator.iosLocator !== undefined) {
+      return `${platformLocator.strategy}${locator.iosLocator.locator}`;
+    } else if (driver.isAndroid && locator.androidLocator !== undefined) {
+      // Explicitly check for driver.isAndroid in case we want to add web tests
+      return `${platformLocator.strategy}${locator.androidLocator.locator}`;
+    }
+    throw new Error('Platform locator is undefined');
+  }
+
+  static async launchApp(bundleId: string): Promise<void> {
+    // Location can be either url for web test dapp or bundleId for native app
+    const platformName = driver.isIOS ? 'IOS' : 'ANDROID';
```

Comment:
take this out of the function so you can use it on all of them without having to "calculate" it each time